### PR TITLE
Cherrypick #1174

### PR DESCRIPTION
**Pull request by mhenriks**
**What this PR does / why we need it**:

Cherrypick #1174 to release 1.13 branch

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Create dedicated SecurityContextConstraint on OpenShift
```

### CODE REVIEW

File: tests/operator_test.go [hunk 1]
```diff
@@ -39,15 +39,37 @@ var _ = Describe("Operator tests", func() {
 		Expect(r.Spec.TLS.Termination).To(Equal(routev1.TLSTerminationReencrypt))
 	})
 
-	It("add cdi-sa to anyuid scc", func() {
-		if !controller.IsOpenshift(f.K8sClient) {
+	It("should create a prometheus service in cdi namespace", func() {
```
Comment (awels) [inline tests/operator_test.go:42]: I think this shouldn't be here, its also causing failures when compiling.

Comment (mhenriks) [reply tests/operator_test.go]: ugh, yeah this was not a clean merge.  Updated.

**Pull request merged**